prov/efa/test/efa_unit_test_mocks.c

```diff
@@ -5,68 +5,128 @@
 #include <setjmp.h>
 #include <cmocka.h>
 #include "efa.h"
+#include "efa_unit_test_mocks.h"
 
-void __real_rxr_pkt_handle_send_completion(struct rxr_ep *ep, struct rxr_pkt_entry *pkt_entry);
+/* mock of rdma-core functions */
+uint8_t g_ibv_gid_mock[EFA_GID_LEN] = {0xfe, 0x80, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x10, 0xcb, 0x06, 0xff, 0xfe, 0x66, 0x83, 0x2f};
 
-void __wrap_rxr_pkt_handle_send_completion(struct rxr_ep *ep, struct rxr_pkt_entry *pkt_entry)
+int g_ibv_create_ah_call_counter = 0;
+struct ibv_ah *efa_mock_ibv_create_ah_increase_call_counter(struct ibv_pd *pd, struct ibv_ah_attr *attr)
 {
-	check_expected(ep);
-	check_expected(pkt_entry);
+	g_ibv_create_ah_call_counter += 1;
+	return  __real_ibv_create_ah(pd, attr);
+}
 
-	if ((int)mock() == 4242) {
-		__real_rxr_pkt_handle_send_completion(ep, pkt_entry);
-	}
+int efa_mock_efadv_query_device_return_mock(struct ibv_context *ibv_ctx,
+					    struct efadv_device_attr *attr,
+					    uint32_t inlen)
+{
+	return mock();
 }
 
-void __real_rxr_pkt_handle_recv_completion(struct rxr_ep *ep,
-										   struct rxr_pkt_entry *pkt_entry,
-										   enum rxr_lower_ep_type lower_ep_type);
+struct ibv_send_wr *g_ibv_send_wr_list_head;
+struct ibv_send_wr *g_ibv_send_wr_list_tail;
 
-void __wrap_rxr_pkt_handle_recv_completion(struct rxr_ep *ep,
-										   struct rxr_pkt_entry *pkt_entry,
-										   enum rxr_lower_ep_type lower_ep_type)
+int efa_mock_ibv_post_send_save_send_wr(struct ibv_qp *qp, struct ibv_send_wr *wr,
```

Comment:
nit This function does not call `mock()`, so consider removing 'mock' from its name. In the context of cmocka, it is tempting to do `will_return(efa_mock_ibv_post_send_save_send_wr, 0)` and later realize it does not do what the name suggests.

Reply:
I don't understand this mock, could you please add a comment?

Reply:
I think the word `mock` is a well accepted term in area of unit testing, to indicate a function that has same signature of a real fucntion, but behave differently. so it should not be confused by the function `mock()`.

Reply:
Added a comment.